test/fail/ok/inf-error-small.tc.ok

```diff
@@ -1,81 +1,62 @@
-inf-error-small.mo:43.13-43.34: type error [M0098], cannot apply function `.get` of type
-  <K, V>(n : K) -> ?V
-to argument of type
+inf-error-small.mo:43.13-43.34: type error [M0096], expression of type
+  ?Text
+cannot produce expected type
+  Text
+inf-error-small.mo:44.13-44.47: type error [M0096], expression of type
+  ?Text
+cannot produce expected type
+  Text
+inf-error-small.mo:46.13-46.45: type error [M0096], expression of type
+  ?Text
+cannot produce expected type
   Text
-because there is no way to satisfy subtyping
-  ?Text  <:  Text  (for the expected return type) 
-inf-error-small.mo:44.13-44.47: type error [M0098], cannot apply function `.get` of type
-  <K, V>(compare : (implicit : (K, K) -> Order), n : K) -> ?V
-to argument of type
-  ((n : Nat, m : Nat) -> Order, Text)
-because there is no way to satisfy subtyping
-  ?Text  <:  Text  (for the expected return type) 
-inf-error-small.mo:46.13-46.45: type error [M0098], cannot apply function `.get` of type
-  <K, V>(compare : (implicit : (K, K) -> Order), n : K) -> ?V
-to argument of type
-  ({map : [(Nat, [var Text])]}, Text)
-because there is no way to satisfy subtyping
-  ?Text  <:  Text  (for the expected return type) 
 inf-error-small.mo:47.26-47.58: type error [M0233], wrong number of arguments: expected 1 but got 3
 Expected 1 argument of type:
   (n : Nat)
 But got 3 arguments of type:
   {map : [(Nat, [var Text])]}
   (n : Nat, m : Nat) -> Order
   Text
-inf-error-small.mo:48.13-48.45: type error [M0098], cannot apply function `.get` of type
-  <K, V>(compare : (implicit : (K, K) -> Order), n : K) -> ?V
-to argument of type
-  ({map : [(Nat, [var Text])]}, Text)
-because there is no way to satisfy subtyping
-  ?Text  <:  Text  (for the expected return type) 
-inf-error-small.mo:50.13-50.29: type error [M0098], cannot apply function `.get` of type
-  <K, V>(n : K) -> ?V
-to argument of type
-  Nat
-because there is no way to satisfy subtyping
-  ?Text  <:  Bool  (for the expected return type) 
-inf-error-small.mo:53.32-53.53: type error [M0098], cannot apply function `.get` of type
-  <K, V>(n : K) -> ?V
-to argument of type
+inf-error-small.mo:48.13-48.45: type error [M0096], expression of type
+  ?Text
+cannot produce expected type
+  Text
+inf-error-small.mo:50.13-50.29: type error [M0096], expression of type
+  ?Text
+cannot produce expected type
+  Bool
+inf-error-small.mo:53.25-53.46: type error [M0230], Cannot determine implicit argument `compare` of type
+  (Any, Any) -> Order
```

Comment:
bimatch succeeds with `K=Any`, which is dumb but valid ofc.
then the check_exp on the hole expression fails because there is no such implicit.

Reply:
this is same on master
```
inf-error-small.mo:53.25-53.46: type error [M0230], Cannot determine implicit argument `compare` of type
  (Any, Any) -> Order
Hint: If you're trying to omit an implicit argument named `compare` you need to have a matching declaration named `compare` in scope.
```